### ISSUE TITLE
matlab2tikz: plotyy overlapping legends 

### ISSUE THREAD

**Issue by grasshoppers** [matlab2tikz/matlab2tikz#140]
I experienced a problem when I created a figure with plotyy, markers and different
 legends. In matlab, all looks quit fine. After compiling the latex pdf, the
 legends overlap. 

**EDIT[Oleg]**  Edited out the long example. Wrote below a MWE.

**Comment by okomarov** [comment]
```matlab
[AX,P1,P2] = plotyy(1:10, [(1:10)', (10:-1:1)'], 1:10, repmat([-5,5],1,5));
lh = legend([P1(1) P1(2) P2],'A','B','C');
set(lh,'Box','off','Orientation','horizontal','Location','NorthOutside');
```
![capture](https://cloud.githubusercontent.com/assets/3731173/5824109/129fadda-a0da-11e4-9835-9fdb2a4054e6.PNG)

**Comment by okomarov** [comment]
What happens is that Matlab uses the legend location option to set the position of the legend in the figure while pgfplots still needs to specify the legend with respect to which axis it belong. Thus, `findLegendInformation()` splits/assigns the entries of the legend by determining to which axis they are subordinated. 

In R2014a/b the legend has an undocumented property called `'Axes'` which would make our life easier and would clean up teh code considerably. I would need to check if this is backwards compatble.

**Comment by PeterPablo** [comment]
:+1:

**Comment by PeterPablo** [comment]
@okomarov I just ran your code on R2014b with 68cc381 and unfortunately the legendentry C is not present. Any clues?

**Comment by okomarov** [comment]
@PeterPablo Yes, this is expected. It's a transition to proper legend implementation for plotyy.

**Comment by okomarov** [comment]
Fixed in #773
